pkg/artifactory/resource_artifactory_user.go

```diff
@@ -144,29 +143,42 @@ func packUser(user User, d *schema.ResourceData) error {
 	}
 
 	if errors != nil && len(errors) > 0 {
-		return fmt.Errorf("failed to pack user %q", errors)
+		return diag.Errorf("failed to pack user %q", errors)
 	}
 
 	return nil
 }
 
-func resourceUserCreate(d *schema.ResourceData, m interface{}) error {
+func resourceUserCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	user := unpackUser(d)
 
-	if user.Name == "" {
-		return fmt.Errorf("user name cannot be empty")
-	}
+	var diags diag.Diagnostics
 
 	if user.Password == "" {
-		return fmt.Errorf("no password supplied. Please use any of the terraform random password generators")
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  "No password supplied",
+			Detail:   "One will be generated (10 characters with 1 digit, 1 symbol, with upper and lower case letters) and this may fail as your Artifactory password policy can't be determined by the provider.",
+		})
+
+		// Generate a password that is 10 characters long with 1 digit, 1 symbol,
+		// allowing upper and lower case letters, disallowing repeat characters.
+		randomPassword, err := password.Generate(10, 1, 1, false, false)
```

Comment:
I guess. It doesn't seem very functional to me. I would have done something like this:

```go
password.Generate(
     min(10),
     digit(1),
     symbol(1),
     upper(),
     lower()
)
```
And then the generator literally just executes each function

But, whatever

Reply:
Take that up with https://github.com/sethvargo/go-password